galsim/gsobject.py

```diff
@@ -1432,63 +1457,104 @@ def drawImage(self, image=None, nx=None, ny=None, bounds=None, scale=None, wcs=N
 
             if prof.isAnalyticX():
-                added_photons = prof.drawReal(draw_image)
+                added_photons = prof.drawReal(draw_image, add)
             else:
-                added_photons = prof.drawFFT(draw_image, wmult)
+                added_photons = prof.drawFFT(draw_image, add, wmult)
 
             if sensor is not None:
                 ud = galsim.UniformDeviate(rng)
                 photons = galsim.PhotonArray.makeFromImage(draw_image, rng=ud)
                 for op in surface_ops:
                     op.applyTo(photons)
-                added_photons = sensor.accumulate(photons, imview)
+                if imview.dtype in [np.float32, np.float64]:
+                    added_photons = sensor.accumulate(photons, imview)
+                else:
+                    # Need a temporary
+                    im1 = galsim.ImageD(bounds=imview.bounds)
+                    added_photons = sensor.accumulate(photons, im1)
+                    imview.array[:,:] += im1.array.astype(imview.dtype, copy=False)
 
         image.added_flux = added_photons / flux_scale
         if save_photons:
             image.photons = photons
 
         return image
 
-    def drawReal(self, image):
+    def drawReal(self, image, add_to_image=False):
         """
         Draw this profile into an Image by direct evaluation at the location of each pixel.
 
         This is usually called from the `drawImage` function, rather than called directly by the
-        user.  In particular, the input image must be already set up with defined bounds with
-        the center set to (0,0).  It also must have a pixel scale wcs.  The profile being
-        drawn should have already been converted to image coordinates via
+        user.  In particular, the input image must be already set up with defined bounds.  The
+        profile will be drawn centered on whatever pixel corresponds to (0,0) with the given
+        bounds, not the image center (unlike drawImage).  The image also must have a PixelScale
+        wcs.  The profile being drawn should have already been converted to image coordinates via
 
             >>> image_profile = original_wcs.toImage(original_profile)
 
-        The image is not cleared out before drawing.  So this profile will be added to anything
-        already on the input image.  This corresponds to `add_to_image=True` in the `drawImage`
-        options.  If you don't want to add to the existing image, just call `image.setZero()`
-        before calling `drawReal`.
-
         Note that the image produced by `drawReal` represents the profile sampled at the center
         of each pixel and then multiplied by the pixel area.  That is, the profile is NOT
         integrated over the area of the pixel.  If you want to render a profile integrated over
-        the pixel, you can convolve with a Pixel first and draw that.
+        the pixel, you can convolve with a Pixel first and draw that.  This is equivalent to
```

Comment:
Thanks. I switched the order of the last two sentences, which I think gets the antecedent correct.